src/test/java/com/tenx/ms/retail/store/rest/StoreControllerTest.java

```diff
@@ -0,0 +1,142 @@
+package com.tenx.ms.retail.store.rest;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.tenx.ms.commons.config.Profiles;
+import com.tenx.ms.retail.BaseRetailIntegrationTest;
+import com.tenx.ms.retail.RetailServiceApp;
+import com.tenx.ms.retail.store.rest.dto.Store;
+import org.apache.commons.io.FileUtils;
+import org.flywaydb.test.annotation.FlywayTest;
+import org.flywaydb.test.junit.FlywayTestExecutionListener;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = {RetailServiceApp.class})
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class, FlywayTestExecutionListener.class})
+@ActiveProfiles(Profiles.TEST_NOAUTH)
+public class StoreControllerTest extends BaseRetailIntegrationTest {
+
+    @Value("classpath:store/store_success.json")
+    private File storeSuccessRequest;
+
+    @Value("classpath:store/store_success_name.json")
+    private File storeSuccessOtherNameRequest;
+
+    @Value("classpath:store/store_failed.json")
+    private File failedRequest;
+
+    @Test
+    @FlywayTest
+    public void testCreateStoreSuccess() {
+        Long storeId = createStore();
+        assertEquals("StoreId does not match", Long.valueOf(1L), storeId);
+    }
+
+    @Test
+    @FlywayTest
+    public void testCreateStoreInvalidName() {
+        try {
+            ResponseEntity<String> response = getStringResponse(String.format(STORES_REQUEST_URI, getBasePath()), FileUtils.readFileToString(failedRequest), HttpMethod.POST);
+            assertEquals("HTTP Status code incorrect for create store.", HttpStatus.PRECONDITION_FAILED, response.getStatusCode());
```

Comment:
@snoz2191 @marielbysoares agree. I also called the get endpoint in my assignment.